src/js_native_api_v8.cc

```diff
@@ -2436,7 +2436,7 @@ napi_status NAPI_CDECL napi_create_reference(napi_env env,
   CHECK_ARG(env, result);
 
   v8::Local<v8::Value> v8_value = v8impl::V8LocalValueFromJsValue(value);
-  if (env->module_api_version <= 8) {
+  if (env->module_api_version != NAPI_VERSION_EXPERIMENTAL) {
```

Comment:
Maybe we should have

```suggestion
  if (env->module_api_version > NAPI_VERSION) {
```

When one compiles an add-on with `NAPI_EXPERIMENTAL`, the version returned by `node_api_module_get_api_version_v1()` is `NAPI_VERSION` (currently 8). So, if an add-on maintainer decides they wish to use experimental features, they can turn on `NAPI_EXPERIMENTAL` to take advantage of otherwise unavailable API signatures, and they can also set `NAPI_VERSION` to something greater than what it is currently. That way, they opt into behaviours that are expected to be released in the next *n* versions, without opting into behaviours that are expected to be released in all future versions.

Reply:
OK, so there's a distinction:

```C
#define NAPI_EXPERIMENTAL
#include <node_api.h>
```

will result in `node_api_module_get_api_version_v1()` reporting 8, whereas

```gyp
{
  ...
  "defines": [ "NAPI_EXPERIMENAL"]
  ...
}
```

will result in `node_api_module_get_api_version_v1()` reporting 2147483647. Experimental signatures will be available in both cases.

Reply:
Hmmm ... returning something other than 8 or `NAPI_VERSION_EXPERIMENTAL` trips up the add-on loader. That's where we added the `static_assert`. I wanna debug that, because I really think it's valuable to be able to express the fact that yes, you wanna track experimental behaviour changes, but not ad infinitum.